lib/rule-tester/rule-tester.js

```diff
@@ -179,6 +183,68 @@ function sanitize(text) {
     );
 }
 
+/**
+ * Define `start`/`end` properties as throwing error.
+ * @param {string} objName Object name used for error messages.
+ * @param {ASTNode} node The node to define.
+ * @returns {void}
+ */
+function defineStartEndAsError(objName, node) {
+    Object.defineProperty(node, "start", {
+        get() {
+            throw new Error(`Use ${objName}.range[0] instead of ${objName}.start`);
+        },
+        configurable: true,
+        enumerable: false
+    });
+    Object.defineProperty(node, "end", {
+        get() {
+            throw new Error(`Use ${objName}.range[1] instead of ${objName}.end`);
+        },
+        configurable: true,
+        enumerable: false
+    });
+}
+
+/**
+ * Define `start`/`end` properties of all nodes of the given AST as throwing error.
+ * @param {ASTNode} ast The root node to errorize `start`/`end` properties.
+ * @param {Object} [visitorKeys] Visitor keys to be used for traversing the given ast.
+ * @returns {void}
+ */
+function defineStartEndAsErrorInTree(ast, visitorKeys) {
+    Traverser.traverse(ast, { visitorKeys, enter: defineStartEndAsError.bind(null, "node") });
+    ast.tokens.forEach(defineStartEndAsError.bind(null, "token"));
+    ast.comments.forEach(defineStartEndAsError.bind(null, "comment"));
```

Comment:
it would cause a perf loss. but I'm fine with it as it will not affect end users. :)

Reply:
It adds a couple of seconds to overall `npm test` time, but I'm not sure if there is a more efficient alternative to warn about using `start`/`end` properties?

If it seems worthwhile, this could be optimized a bit to avoid creating some new objects per node/token, e.g., all getters with the same message could share the same function instance (furthermore, we could create just 2-3 property descriptor objects and reuse them).

Reply:
I'm okay with eating a few seconds in test suites but I do think we should make it clear to plugin authors who use `RuleTester` why this change was made, as well as make it clear that this was a conscious trade off. Improvements always welcome, of course!